src/main/java/net/dv8tion/jda/internal/entities/channel/concrete/detached/DetachedForumChannelImpl.java

```diff
@@ -0,0 +1,271 @@
+    @Override
+    public SortedSnowflakeCacheViewImpl<ForumTag> getAvailableTagCache()
+    {
+        return tagCache;
+    }
+
+    @Override
+    public TLongObjectMap<PermissionOverride> getPermissionOverrideMap()
+    {
+        throw detachedException();
+    }
+
+    @Nonnull
+    @Override
+    public ChannelInteractionPermissions getInteractionPermissions()
+    {
+        return interactionPermissions;
+    }
+
+    @Override
+    public boolean isNSFW()
+    {
+        return nsfw;
+    }
+
+    @Override
+    public int getPositionRaw()
+    {
+        return position;
+    }
+
+    @Override
+    public long getParentCategoryIdLong()
+    {
+        return parentCategoryId;
+    }
+
+    @Override
+    public int getSlowmode()
+    {
+        return slowmode;
+    }
+
+    @Override
+    public String getTopic()
+    {
+        return topic;
+    }
+
+    @Override
+    public EmojiUnion getDefaultReaction()
+    {
+        return (EmojiUnion) defaultReaction;
+    }
+
+    @Override
+    public int getDefaultThreadSlowmode()
+    {
+        return defaultThreadSlowmode;
+    }
+
+    @Nonnull
+    @Override
+    public SortOrder getDefaultSortOrder()
+    {
+        return SortOrder.fromKey(defaultSortOrder);
+    }
+
+    @Nonnull
+    @Override
+    public Layout getDefaultLayout()
+    {
+        return Layout.fromKey(defaultLayout);
+    }
+
+    public int getRawFlags()
+    {
+        return flags;
+    }
+
+    public int getRawSortOrder()
+    {
+        return defaultSortOrder;
+    }
+
+    public int getRawLayout()
+    {
+        return defaultLayout;
+    }
+
+    // Setters
+
+    @Override
+    public DetachedForumChannelImpl setParentCategory(long parentCategoryId)
+    {
+        this.parentCategoryId = parentCategoryId;
+        return this;
+    }
+
+    @Override
+    public DetachedForumChannelImpl setPosition(int position)
+    {
+        this.position = position;
+        return this;
+    }
+
+    @Override
+    public DetachedForumChannelImpl setDefaultThreadSlowmode(int defaultThreadSlowmode)
+    {
+        this.defaultThreadSlowmode = defaultThreadSlowmode;
+        return this;
+    }
+
+    @Override
+    public DetachedForumChannelImpl setNSFW(boolean nsfw)
+    {
+        this.nsfw = nsfw;
+        return this;
+    }
+
+    @Override
+    public DetachedForumChannelImpl setSlowmode(int slowmode)
+    {
+        this.slowmode = slowmode;
+        return this;
+    }
+
+    @Override
+    public DetachedForumChannelImpl setTopic(String topic)
+    {
+        this.topic = topic;
+        return this;
+    }
+
+    @Override
+    public DetachedForumChannelImpl setFlags(int flags)
+    {
+        this.flags = flags;
+        return this;
+    }
+
+    @Override
+    public DetachedForumChannelImpl setDefaultReaction(DataObject emoji)
+    {
+        if (emoji != null && !emoji.isNull("emoji_id"))
+            this.defaultReaction = new CustomEmojiImpl("", emoji.getUnsignedLong("emoji_id"), false);
```

Comment:
why is name an empty string? do we not have the data to do this correctly here?

Reply:
Yep:
```
"default_reaction_emoji" : {
    "emoji_id" : "1292121888994230375",
    "emoji_name" : null
}
```